tests/unit/test_reservation.py

```diff
@@ -649,6 +657,254 @@ def test___no_reservation_order___get_connections___connections_returned_in_defa
     ]
 
 
+def test___multiplexer_session_not_created___get_connection_with_multiplexer___raises_type_error(
+    session_management_client: Mock,
+) -> None:
+    with ExitStack() as stack:
+        grpc_session_infos = create_nifake_session_infos(1)
+        grpc_multiplexer_session_infos = create_nimultiplexer_session_infos(1)
+        grpc_session_infos[0].channel_mappings.add(
+            pin_or_relay_name="Pin1",
+            site=2,
+            channel="3",
+            multiplexer_resource_name="Mux0",
+            multiplexer_route="route1",
+        )
+        reservation = MultiSessionReservation(
+            session_management_client, grpc_session_infos, grpc_multiplexer_session_infos
+        )
+        _ = stack.enter_context(reservation.initialize_session(_construct_session, "nifake"))
+
+        with pytest.raises(TypeError) as exc_info:
+            _ = reservation.get_connection_with_multiplexer(object, fake_driver.Session)
+
+        assert "Incorrect type for multiplexer session 'MyMultiplexer0'." in exc_info.value.args[0]
+        assert (
+            "Expected <class 'tests.utilities.fake_driver.Session'>, got <class 'NoneType'>."
+            in exc_info.value.args[0]
+        )
+
+
+def test___multiplexer_session_not_created___get_connection_with_multiplexer_using_multiplexer_session_type_as_object___returns_connection(
```

Comment:
This is kind of wordy. I like the wording "using object type" of the plural version better.

```suggestion
def test___multiplexer_session_not_created___get_connection_with_multiplexer_using_object_type___returns_connection(
```

Reply:
Updated.